contracts/verifying/VerifyingSingletonPaymasterV2.sol

```diff
@@ -1,9 +1,363 @@
+        address oldSigner = verifyingSigner;
+        assembly {
+            sstore(verifyingSigner.slot, _newVerifyingSigner)
+        }
+        emit VerifyingSignerChanged(oldSigner, _newVerifyingSigner, msg.sender);
+    }
+
+    function setUnaccountedEPGasOverhead(
+        uint256 value
+    ) external payable onlyOwner {
+        uint256 oldValue = unaccountedEPGasOverhead;
+        unaccountedEPGasOverhead = value;
+        emit EPGasOverheadChanged(oldValue, value);
+    }
+
+    function setFixedPriceMarkup(uint32 _markup) external payable onlyOwner {
+        require(_markup <= PRICE_DENOMINATOR * 2, "markup too high");
+        uint32 oldValue = fixedPriceMarkup;
+        fixedPriceMarkup = _markup;
+        emit FixedPriceMarkupChanged(oldValue, _markup);
+    }
+
+    /**
+     * @dev This method is called by the off-chain service, to sign the request.
+     * It is called on-chain from the validatePaymasterUserOp, to validate the signature.
+     * @notice That this signature covers all fields of the UserOperation, except the "paymasterAndData",
+     * which will carry the signature itself.
+     * @return hash we're going to sign off-chain (and validate on-chain)
+     */
+    function getHash(
+        UserOperation calldata userOp,
+        address paymasterId,
+        uint48 validUntil,
+        uint48 validAfter,
+        uint32 priceMarkup
+    ) public view returns (bytes32) {
+        //can't use userOp.hash(), since it contains also the paymasterAndData itself.
+        return
+            keccak256(
+                abi.encode(
+                    userOp.getSender(),
+                    userOp.nonce,
+                    keccak256(userOp.initCode),
+                    keccak256(userOp.callData),
+                    userOp.callGasLimit,
+                    userOp.verificationGasLimit,
+                    userOp.preVerificationGas,
+                    userOp.maxFeePerGas,
+                    userOp.maxPriorityFeePerGas,
+                    block.chainid,
+                    address(this),
+                    paymasterId,
+                    validUntil,
+                    validAfter,
+                    priceMarkup
+                )
+            );
+    }
+
+    // Note: do not use this in validation phase
+    function getGasPrice(
+        uint256 maxFeePerGas,
+        uint256 maxPriorityFeePerGas
+    ) internal view returns (uint256) {
+        if (maxFeePerGas == maxPriorityFeePerGas) {
+            //legacy mode (for networks that don't support basefee opcode)
+            return maxFeePerGas;
+        }
+        return minuint256(maxFeePerGas, maxPriorityFeePerGas + block.basefee);
+    }
+
+    // Review below helpers
+    function minuint256(uint256 a, uint256 b) internal pure returns (uint256) {
+        return a < b ? a : b;
+    }
+
+    function maxuint256(uint256 a, uint256 b) internal pure returns (uint256) {
+        return a > b ? a : b;
+    }
```

Comment:
```solidity
    assembly {
        result := xor(a, mul(xor(a, b), gt(b, a)))
    }
```

This is a more gas-efficient alternative that can be replicated in other methods too

Reply:
`returns(uint256 result)`

Reply:
I need max and min util functions for numbers of two types uin256 and uint48. currently I have had to make two different util functions so max and min total 4. Is there a way to make it irrespective of types. @ankurdubey521 

@Aboudjem same I can do for a and b which is uint48?

what would be assembly function for min?